cxf-soap/src/main/java/org/acme/cxf/soap/service/ContactService.java

```diff
@@ -16,13 +16,17 @@
  */
 package org.acme.cxf.soap.service;
 
+import jakarta.jws.WebService;
+
+@WebService(name = "ContactService", serviceName = "ContactService", targetNamespace = ContactService.TARGET_NS)
 public interface ContactService {
+    public static final String TARGET_NS = "http://camel.apache.org/test/ContactService";
 
     void addContact(Contact contact);
 
     Contact getContact(String name) throws NoSuchContactException;
 
-    GetContactsResponse getContacts();
+    Contacts getContacts();
```

Comment:
Why we cannot return `List<Contact>` ?

Reply:
IIRC, Lists do not work well in general in JAX-WS without these wrappers. Arrays seem to be better supported. But when returning `Contact[]` from getContacts(), a bug pops up caused by the fact that Camel or CXF replaces a single element array by the element. Not sure how to workaround it. Maybe some binding config or transformers.

Reply:
Ok thank you.